src/Compilers/Core/Portable/StrongName/PortableStrongNameProvider.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Cci;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class PortableStrongNameProvider : StrongNameProvider
+    {
+        private readonly ImmutableArray<string> _keyFileSearchPaths;
+        internal StrongNameFileSystem FileSystem { get; private set; }
+
+        public PortableStrongNameProvider(ImmutableArray<string> keySearchPaths = default, StrongNameFileSystem ioOp = default)
+        {
+            FileSystem = ioOp ?? new StrongNameFileSystem();
+            _keyFileSearchPaths = keySearchPaths.NullToEmpty();
+        }
+
+        public override bool Equals(object other)
```

Comment:
Should `Equals` compare `_keyFileSearchPaths` and `FileSystem` as well?

Where are we comparing `StrongNameProvider` instances? Only in `CompilationOptions`? If so, can we simply compare using `ReferenceEquals` in that case? Then all of the various `Equals` implementations can be removed.